src/screens/Observation/EditObservation.js

```diff
@@ -0,0 +1,398 @@
+import PropTypes from "prop-types";
+import {
+  Text,
+  ScrollView,
+  KeyboardAvoidingView,
+  Picker,
+  Image,
+  View
+} from "react-native";
+
+import { styles } from "./styles";
+import { Button, Input } from "../../components";
+import { translate } from "../../i18n";
+import backIcon from "../../assets/arrow-pointing-to-left.png";
+import photoPlaceholder from "../../assets/photoPlaceholder.png";
+
+const pickerValueType = PropTypes.shape({
+  label: PropTypes.string,
+  value: PropTypes.string
+});
+const pickerValuesArrayType = PropTypes.arrayOf(pickerValueType);
+
+const getDescriptionBlock = (headerText, descriptionText) => (
+  <View>
+    <Text style={styles.headerText}>{headerText}</Text>
+    <Text style={styles.descriptionText}>{descriptionText}</Text>
+  </View>
+);
+
+const getEmptyPickerValue = () => {
+  return {
+    label: "",
+    value: ""
+  };
+};
+
+const EditObservation = props => {
+  const {
+    birdSpeciesDefault,
+    birdSpeciesValues,
+    birdSexDefault,
+    birdSexValues,
+    birdAgeDefault,
+    birdAgeValues,
+    birdObstaclesDefault,
+    birdObstaclesValues,
+    ringsDefaultValues,
+    ringTypeValues,
+    ringMaterialValues,
+    ringColorValues,
+    ringLocationValues,
+    birdPhotos,
+    countryDefault,
+    countryValues,
+    regionDefault,
+    coordinatesDefault,
+    commentDefault,
+    dateTimeDefault,
+    dateTimeInaccuracyDefault
+  } = props;
+  const [
+    {
+      birdSpecies,
+      birdSex,
+      birdAge,
+      birdObstacles,
+      country,
+      region,
+      coordinates,
+      comment,
+      dateTime,
+      dateTimeInaccuracy
+    },
+    setFieldValue
+  ] = useState({
+    birdSpecies: birdSpeciesDefault,
+    birdSex: birdSexDefault,
+    birdAge: birdAgeDefault,
+    birdObstacles: birdObstaclesDefault,
+    country: countryDefault,
+    region: regionDefault,
+    coordinates: coordinatesDefault,
+    comment: commentDefault,
+    dateTime: dateTimeDefault,
+    dateTimeInaccuracy: dateTimeInaccuracyDefault
+  });
+
+  const { rings, setRingsValues } = useState(ringsDefaultValues);
+
+  const onSubmitPress = () => {
+    props.onSubmit();
+  };
+  const onBackPress = () => {
+    props.onBackNavigation();
+  };
+  const onCurrentPositionPress = () => {
+    props.onCurrentPosition();
+  };
+  const onSearchOnMapPress = () => {
+    props.onSearchOnMap();
+  };
+  const onCurrentDateTimePress = () => {
+    props.onCurrentDateTime();
+  };
+
+  const getNewRingId = () => `${Object.keys(rings).length + 1}`;
+  const getEmptyRingFieldsGroup = () => {
+    return {
+      type: getEmptyPickerValue(),
+      material: getEmptyPickerValue(),
+      color: getEmptyPickerValue(),
+      location: getEmptyPickerValue(),
+      tagId: ""
+    };
+  };
+  const onAddOneMoreRing = () => {
+    setRingsValues(
+      Object.assign({}, rings, {
+        [getNewRingId()]: getEmptyRingFieldsGroup()
+      })
+    );
+  };
+
+  return (
+    <View>
```

Comment:
In my opinion return block is too big. Can we break it in separate functions and just call them inside?

Reply:
Done